Add a GitHub pull request template

lcolladotor opened the pull request:
This pull request resolves https://github.com/Bioconductor/BiocCheck/issues/86.


As part of the commit, a new file `.github/pull_request_template.md` was created that contains a template file based on the [GitHub instructions](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository) and the top lines at [the `biocthis` issue template](https://github.com/lcolladotor/biocthis/blob/master/.github/ISSUE_TEMPLATE/issue_template.md).

Ideally, this file should be properly recognized by GitHub.

The contents of the md file rely heavily on the presentation by Lori Shepard for the BiocCheck-a-thon from May 18th 2020. The contents on the presentation are available at
[https://docs.google.com/presentation/d/1DkN2WVPOMVGqUtlSSrWbx6IMjtGP_cEHoE3nfOEnD68/edit#slide=id.p](here).

This PR also adds the `.github` directory to the `.Rbuildignore`. This can clash with other PRs if they also added this line to the `.Rbuildignore` file.

Best,
Leo

lcolladotor committed:
Add a GitHub pull request template

This commit is related to https://github.com/Bioconductor/BiocCheck/issues/86.
As part of the commit, a new file `.github/pull_request_template.md` was created
that contains a template file based on the instructions at
https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository
and the top lines at https://github.com/lcolladotor/biocthis/blob/master/.github/ISSUE_TEMPLATE/issue_template.md

Ideally, this should be properly recognized by GitHub.

The contents of the md file rely heavily on the presentation by Lori Shepard
for the BiocCheck-a-thon from May 18th 2020. The contents on the presentation
are available at
https://docs.google.com/presentation/d/1DkN2WVPOMVGqUtlSSrWbx6IMjtGP_cEHoE3nfOEnD68/edit#slide=id.p

mtmorgan commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,20 @@
+* [ ] Update the NEWS file (required)
+* [ ] Update the Vignette file (required)
+* [ ] Add unit tests (optional)
+* [ ] Add a flag to turn off higher level checks (optional)
```
probably this is too specific for a generic pull request; I think it should not be mentioned

mtmorgan commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,20 @@
+Prior to sending the pull request, please verify that `R CMD build` and `R CMD check` run without warnings or errors on the latest Bioconductor-devel (currently in May 2020 that would be Bioconductor 3.12) and complete the following steps:
+
+* [ ] Update the NEWS file (required)
+* [ ] Update the Vignette file (required)
```
trivial but please lower-case 'vignette'

mtmorgan reviewed (commented):
this looks promising!

lcolladotor committed:
Respond to the pull request review comments

This commit responds to the pull request review comments from https://github.com/Bioconductor/BiocCheck/pull/87#pullrequestreview-414581029. In particular, it:

* Removes a checkbox that was too specific.
* Fixes the spelling of 'vignette'.

Furthermore, this commit:

* Updates the NEWS file
* Adds a short entry to the vignette file.

lcolladotor commented:
Hi,

Thank you for your review comments. I just sent a [new commit](https://github.com/Bioconductor/BiocCheck/pull/87/commits/996291466a8f2a704ab1da08fe975d919dec6bb3) that addresses the review comments but also edits the NEWS and vignette files.

Best,
Leo

LiNk-NY commented:
@lcolladotor 
Thank you for sending this in. 
This is a GitHub-side change and it doesn't really affect the package. For this reason, I don't think the NEWS file should be updated.
